src/pdf/generator.py

```
@@ -0,0 +1,130 @@
+            pdf.docinfo["/Title"] = title
+            pdf.docinfo["/Creator"] = "kindle2pdf"
+            pdf.docinfo["/Producer"] = "kindle2pdf with img2pdf"
+            
+            # Save with linearization for fast web view
+            pdf.save(temp_path, linearize=True)
+        
+        logger.info(f"Generated PDF: {temp_path}")
+        return temp_path
+        
+    except Exception as e:
+        raise PDFError(f"Failed to generate PDF: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)